framework_agreement/model/framework_agreement.py

```diff
@@ -268,10 +272,10 @@ def _compute_state(self):
 
     @api.multi
     @api.depends('quantity',
-                 'purchase_order_ids.framework_agreement_id',
-                 'purchase_order_ids.state',
-                 'purchase_order_ids.order_line',
-                 'purchase_order_ids.order_line.product_qty')
+                 'purchase_line_ids.framework_agreement_id',
+                 'purchase_line_ids.order_id.state',
+                 'purchase_line_ids',
+                 'purchase_line_ids.product_qty')
```

Comment:
Those triggers seems strange compared to `_compute_state` method.

Aren't `state`, `start_date`, `end_date` missing?

About computed fields `available_quantity` I guess it doesn't aggregate dependencies of dependencies? You could add a comment to explain why we keep all those dependences?
like `# dependencies of available_quantity`